### PR TITLE
Feature/sort/unlimited color

### PR DESCRIPTION
**Pull request by termitehuegel**

### CODE REVIEW

File: src/main/algorithms/sort/sortSimulationStepFactory.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import type { SortSimulationStep } from '@/main/algorithms/sort/types'
+import { SortColor } from '@/main/algorithms/sort/types'
+
+export class SortSimulationStepFactory {
+
+    static createHighlightedSimulationStep(sortedValues: number[], highlightedIndices: {color: string, index: number}[]): SortSimulationStep  {
+        return {
+            sortedValues: sortedValues.map((value, index) => {
+                return {value: value, displayColor: highlightedIndices.find(highlightedIndex => highlightedIndex.index == index)?.color ?? SortColor.NEUTRAL}
```
Comment (yxyx-github) [inline src/main/algorithms/sort/sortSimulationStepFactory.ts:9]: Formatting: keep one blank inside curly brackets everywhere, e.g.: { value: ..., displayColor: ... }

File: src/main/algorithms/sort/quickSort.ts [hunk 2]
```diff
@@ -1,26 +1,23 @@
 import type {
-    HighlightedIndex,
     SortAlgorithmImplementation,
     SortSimulation,
     SortSimulationStep
 } from '@/main/algorithms/sort/types'
 import { ProtocolBuilder } from '@/main/simulation/protocolBuilder'
 import type { TrackableProgress } from '@/main/progressTracker/types'
+import {SortSimulationStepFactory} from '@/main/algorithms/sort/sortSimulationStepFactory'
```
Comment (yxyx-github) [inline src/main/algorithms/sort/quickSort.ts:8]: Formatting: keep one blank inside curly brackets.

File: src/main/algorithms/sort/insertionSort.ts [hunk 3]
```diff
@@ -1,15 +1,14 @@
 import { ProtocolBuilder } from '@/main/simulation/protocolBuilder'
-import type { HighlightedIndex, SortAlgorithmImplementation, SortSimulation, SortSimulationStep } from '@/main/algorithms/sort/types'
+import type {SortAlgorithmImplementation, SortSimulation, SortSimulationStep } from '@/main/algorithms/sort/types'
```
Comment (yxyx-github) [inline src/main/algorithms/sort/insertionSort.ts:2]: Formatting: insert one blank

File: src/main/algorithms/sort/heapSort.ts [hunk 4]
```diff
@@ -107,4 +82,41 @@ export class HeapSort implements SortAlgorithmImplementation {
          seinen Kindern verglichen und getauscht, sofern es sich nicht um das größte bzw. kleinste Element handelt. Dieser
          Vorgang wird rekursiv verkettet, so dass auch die Unterbäume Max- bzw. Min-Heaps sind.`]
     }
+
```
Comment (yxyx-github) [inline src/main/algorithms/sort/heapSort.ts:85]: Formatting: move methods above description method

File: src/main/algorithms/sort/bubbleSort.ts [hunk 5]
```diff
@@ -40,13 +36,11 @@ export class BubbleSort implements SortAlgorithmImplementation {
     }
 
     private createStep(values: number[], pointer: number, lastElement: number): SortSimulationStep {
```
Comment (yxyx-github) [inline src/main/algorithms/sort/bubbleSort.ts:38]: Formatting: move method above description

File: src/components/app/simulation/visualization/SortVisualization.vue [hunk 6]
```diff
@@ -4,46 +4,33 @@
...
-    ...props.colors,
-}))
 
-function fill(index: number): string {
-    const highlightType: HighlightColor = props.step.highlightedIndices.find(highlightedIndex => highlightedIndex.index === index)?.type ?? 'neutral'
-    return mergedColors.value[highlightType]
-}
-</script>
 
+</script>
```
Comment (yxyx-github) [inline src/components/app/simulation/visualization/SortVisualization.vue:34]: Formatting: keep one blank line between </script> and <style>

**Review by yxyx-github — CHANGES_REQUESTED** [review]

File: src/main/algorithms/sort/sortSimulationStepFactory.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import type { SortSimulationStep } from '@/main/algorithms/sort/types'
+import { SortColor } from '@/main/algorithms/sort/types'
+
+export class SortSimulationStepFactory {
+
+    static createHighlightedSimulationStep(sortedValues: number[], highlightedIndices: {color: string, index: number}[]): SortSimulationStep  {
```
[the remark this answers is not in the document]
Comment (termitehuegel) [reply src/main/algorithms/sort/sortSimulationStepFactory.ts:6]: This does not make sense in the context of the issue.
We want to allow all Colors but SortColor only contains constants with often used colors.

**Review by yxyx-github — APPROVED** [review]

**Pull request merged**